### PR TITLE
[do not merge] CLI commands for AzureDiskEncryption

### PR DESCRIPTION
**Pull request by SudhakaraReddyEvuri**
This change list contains CLI commands for AzureDiskEncryption. This adds the below commands
1. azure vm enable-disk-encryption 
2. azure vm show-disk-encryption-status

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@SudhakaraReddyEvuri__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

File: lib/commands/arm/vm/vmClient._js [hunk 1]
```diff
@@ -1169,7 +1169,153 @@ __.extend(VMClient.prototype, {
...
+    var extensionResult = serviceClients.computeManagementClient.virtualMachineExtensions.getWithInstanceView(resourceGroupName, name, vmConstants.EXTENSIONS.AZURE_DISK_ENCRYPTION_WINDOWS_EXTENSION_NAME, _);
+    if(!extensionResult ||
+       !extensionResult.virtualMachineExtension || 
+       !utils.ignoreCaseEquals(extensionResult.virtualMachineExtension.publisher , vmConstants.EXTENSIONS.AZURE_DISK_ENCRYPTION_WINDOWS_EXTENSION_PUBLISHER) ||
+       !utils.ignoreCaseEquals(extensionResult.virtualMachineExtension.extensionType , vmConstants.EXTENSIONS.AZURE_DISK_ENCRYPTION_WINDOWS_EXTENSION_NAME) ||
+       !utils.ignoreCaseEquals(extensionResult.virtualMachineExtension.provisioningState , vmConstants.EXTENSIONS.EXTENSION_PROVISIONING_SUCCEEDED))
+    {
+      throw new Error(util.format($('Virtual machine "%s" extension with successful status not found after installing extension  '), vmConstants.EXTENSIONS.AZURE_DISK_ENCRYPTION_WINDOWS_EXTENSION_NAME));
+    }
+    var statusUrl = extensionResult.virtualMachineExtension.instanceView.statuses[0].message;
```
Comment (huangpf) [inline lib/commands/arm/vm/vmClient._js]: How about null or empty list checking on ```statuses```?

File: lib/commands/arm/vm/vmExtensionProfile._js [hunk 2]
```diff
@@ -647,7 +647,68 @@ __.extend(VMExtensionProfile.prototype, {
...
+
+  _generateAzureDiskEncryptionWindowsProfile: function() {
+    var self = this;
+
+    //Check parameters
+
+    if ((utils.stringIsNullOrEmpty(this.params.aadClientId)) || 
+        (utils.stringIsNullOrEmpty(this.params.diskEncryptionKeyVaultUrl)) ||
+        (utils.stringIsNullOrEmpty(this.params.diskEncryptionKeyVaultId))) {
+      throw new Error($('AadClientID, DiskEncryptionKeyVaultUrl, DiskEncryptionKeyVaultId are mandatory inputs'));
```
Comment (huangpf) [inline lib/commands/arm/vm/vmExtensionProfile._js]: I guess it's better to use the real CLI parameter name format like ```--aad-client-id```, etc.

File: lib/commands/arm/vm/vmExtensionProfile._js [hunk 3]
```diff
@@ -647,7 +647,68 @@ __.extend(VMExtensionProfile.prototype, {
...
+    }
+
+    if ((utils.stringIsNullOrEmpty(this.params.aadClientSecret)) && 
+        (utils.stringIsNullOrEmpty(this.params.aadClientCertThumbprint))){
+      throw new Error($('Please provide AadClientSecret or AadClientCertThumbprint correspondong to AadClientID'));
```
Comment (huangpf) [inline lib/commands/arm/vm/vmExtensionProfile._js]: same.

File: lib/commands/arm/vm/vmExtensionProfile._js [hunk 4]
```diff
@@ -647,7 +647,68 @@ __.extend(VMExtensionProfile.prototype, {
...
+    }
+
+    if (!utils.stringIsNullOrEmpty(this.params.keyEncryptionKeyUrl) &&
+        (utils.stringIsNullOrEmpty(this.params.keyEncryptionKeyVaultId))){
+      throw new Error($('Please provide KeyEncryptionKeyVaultId correspondong to KeyEncryptionKeyUrl'));
```
Comment (huangpf) [inline lib/commands/arm/vm/vmExtensionProfile._js]: same.

File: lib/commands/arm/vm/vmClient._js (hunk 1 above)
Comment (SudhakaraReddyEvuri) [inline lib/commands/arm/vm/vmClient._js]: Resolved it

**Comment by huangpf** [comment]
@yugangw-msft @amarzavery 
This PR from the disk encryption team needs to be taken for the current release.

**Comment by yugangw-msft** [comment]
@huangpf, we already cut a release branch, and have built out installers. This PR has been marked as "Do not merge" and still failed CI.
So recommend for next release. Let us know if this is a crucial thing, which involves the work from your side of resetting the PR to the releasing branch, get CI passing, get reviewed, and handle the test reset. 
It appears too late.

**Pull request merged**